### ISSUE TITLE
Popup a11y: Unable to trap screen reader focus

### ISSUE THREAD

**Issue by PahaN47** [gravity-ui/uikit#1746]
We have received a lot of a11y feedback regarding the `Popup` component and others that use it. The issue seems to be that while we are able to set `focusTrap` and `autoFocus` to `true` and it does trap focus to some extent with `Tab` presses not leaving the overlay element this is not enough.
Most users that need accessible features traverse the page using a screen reader program that usually provides its own interface for interaction with web content. Current implementation of `focusTrap` does not account for this, so in practice for these users focus does not get trapped which appears to be somewhat confusing for them.
There is a simple solution that I propose should be implemented in some way. Setting `role="dialog"` with `aria-modal="true"` should do the trick in most cases. Though it is not a universal solution (e.g. with VoiceOver it works in Safari, but not in Firefox), it seems to be the intended one.
Here is a Codesandbox fork with a [rough demonstration](https://codesandbox.io/p/github/PahaN47/gravity-ui-uikit/main) of the difference between the current implementation and an accessible one (`Components -> Overlays -> Popup -> A 11 Y Example`).

**Comment by ValeraS** [comment]
Here is note from https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/
> Because marking a dialog modal by setting [aria-modal](https://w3c.github.io/aria/#aria-modal) to true can prevent users of some assistive technologies from perceiving content outside the dialog, users of those technologies will experience severe negative ramifications if a dialog is marked modal but does not behave as a modal for other users. So, mark a dialog modal only when both:
> 1. Application code prevents all users from interacting in any way with content outside of it.
> 1. Visual styling obscures the content outside of it.

The popup doesn't visually hide the content outside of it like a Modal does, so setting `aria-modal="true"` on it seems wrong.

To set the "dialog" role, you can use the "role" property.
